quickannotator/api/v1/annotation/utils.py

```diff
@@ -0,0 +1,90 @@
+#%%
+import numpy as np
+import ray
+import time
+import os
+
+from quickannotator.db import get_session
+from quickannotator.db.utils import FileSystemManager, build_annotation_table_name
+from quickannotator.db.crud.annotation import AnnotationStore
+from quickannotator.dsa_sdk import DSAClient
+import geojson
+from quickannotator import constants
+from quickannotator.db.crud.image import get_image_by_id
+
+
+@ray.remote
+class ProgressTracker:
+    def __init__(self, total):
+        self.total = total
+        self.progress = 0
+
+    def increment(self):
+        self.progress += 1
+
+    def get_progress(self):
+        return (self.progress / self.total) * 100
+
+
+@ray.remote
+class AnnotationExporter:
+    def __init__(self, image_ids, annotation_class_ids, progress_actor):
+        image_ids_grid, annotation_class_ids_grid = np.meshgrid(image_ids, annotation_class_ids, indexing='ij')
+        self.id_pairs = list(zip(image_ids_grid.ravel(), annotation_class_ids_grid.ravel()))
+        self.progress_actor = progress_actor
+
+    def export_to_dsa(self, api_uri, api_key, folder_id):
+        client = DSAClient(api_uri, api_key)
+        user = client.get_user_by_token()
+        if user is None:
+            raise Exception("Failed to get user ID from token")
+        
+        user_id = user['_id']
+
+        for image_id, annotation_class_id in self.id_pairs:
+            with get_session() as db_session:
+                image_name = get_image_by_id(image_id).name
+                dsa_item = client.get_item_by_name(folder_id, image_name)
+                if dsa_item is None:
+                    raise Exception(f"Item with name {image_name} not found in folder {folder_id}")
+                
+                dsa_item_id = dsa_item['_id']
+                store = AnnotationStore(image_id, annotation_class_id, True, False)
+                feature_collection = store.get_all_annotations_as_feature_collection()    # TODO: this should probably be done with tempfile to avoid memory issues
```

Comment:
indeed - this whole thing should be done either via tempfiles, or in a streaming manner, otherwise we get in trouble pretty quickly if there is a large data class for export :-\

Reply:
so this part here still will generate a whole file in memory first before transmitting, is that correct? is there an opportunity to do this with something like a generator?